src/vs/editor/contrib/wordHighlighter/browser/wordHighlighter.ts

```diff
@@ -676,6 +695,22 @@ class WordHighlighter {
 		}
 	}
 
+	private computeWithModel(model: ITextModel, selection: Selection, word: IWordAtPosition | null, otherModels: ITextModel[]): IOccurenceAtPositionRequest | null {
+		if (!otherModels.length) {
+			return computeOccurencesAtPosition(this.providers, model, selection, word, this.editor.getOption(EditorOption.wordSeparators));
+		} else {
+			return computeOccurencesMultiModel(this.multiDocumentProviders, model, selection, word, this.editor.getOption(EditorOption.wordSeparators), otherModels);
+		}
+	}
+
+	private computeWithQuery(query: IWordHighlighterQuery, otherModels: ITextModel[]): IOccurenceAtPositionRequest | null {
+		if (!otherModels.length) {
+			return computeOccurencesAtPosition(this.providers, query.modelInfo!.model, query.modelInfo!.selection, query.word, this.editor.getOption(EditorOption.wordSeparators));
+		} else {
+			return computeOccurencesMultiModel(this.multiDocumentProviders, query.modelInfo!.model, query.modelInfo!.selection, query.word, this.editor.getOption(EditorOption.wordSeparators), otherModels);
```

Comment:
In `computeWithQuery`,  the `query.modelInfo` can potentionally be `null`, right? I'm not sure if we can safely do `query.modelInfo!`.